src/enumerator.rs

```diff
@@ -132,23 +142,23 @@ impl<'a> Enumerator<'a> {
         }));
 
         Ok(Devices {
-            enumerator: self,
+            enumerator: self.clone(),
```

Comment:
If I understand this correctly, this could introduce memory errors, because it allows more than one `Devices` iterator to access the enumerator simultaneously. Since the iterator is no longer seen to borrow from the enumerator, that allows the enumerator (and thus the device list) to by mutated while accessing the device list.

https://github.com/systemd/systemd/blob/90f989861e1f7fd4465a8dddd1721b54ecb3f273/src/libudev/libudev-enumerate.c#L126-L164